### PR TITLE
Added blocking bubble dome at start of round when choosing blocks.

### PR DESCRIPTION
**Pull request by salberico**
#64

### CODE REVIEW

File: Assets/Scripts/RedRunner/GameManager.cs [hunk 1]
```diff
@@ -302,6 +305,25 @@ public void RespawnCharacter(Character character)
             character.Reset();
         }
 
+        public void LockCharacterToStart()
+        {
+            GameObject respawn = SpawnSingleton.instance;
+            Vector3 position = respawn.transform.position;
+            Debug.Log("Text: " + position);
+            position.y += 2.56f;
+            float width = respawn.GetComponent<SpriteRenderer>().bounds.size.x;
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/GameManager.cs]: width -> diameter for clarity if we're keeping the circle? or just a comment somewhere about it

File: Assets/Scripts/RedRunner/Networking/RoundsManager.cs [hunk 2]
```diff
@@ -48,6 +48,19 @@ public void RpcResetRound()
             GameManager.Singleton.StartGame();
             if (!Application.isBatchMode) {
                 GameManager.Singleton.RespawnMainCharacter();
+                GameManager.Singleton.LockCharacterToStart();
+                ServerSpawner.Instance.InitiateChoosing();
+            }
+        }
+
+        // receive on client to reset round
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/Networking/RoundsManager.cs]: receive on client to reset round -> "received on client to unblock character when choosing phase is over" or something along those lines

File: Assets/Scripts/RedRunner/TerrainGeneration/SpawnLock.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    public void Finish()
+    {
+        Destroy(gameObject);
+        destroySpawnLock.UnregisterAction(finishAction);
+    }
+
+    public void SetWidth(float width)
+    {
+        float scale = width / 2.56f;
```
Comment (WilsonCWu) [inline Assets/Scripts/RedRunner/TerrainGeneration/SpawnLock.cs]: maybe its time to make this an actual constant somewhere?

**Comment by WilsonCWu** [comment]
Is it worthwhile to reset the player position before starting the game? not a big deal either way

**Comment by salberico** [comment]
#64

> 
> 
> Is it worthwhile to reset the player position before starting the game? not a big deal either way

I believe this is the case already

**Review by cd9 — COMMENTED** [review]
lgtm but not sure why we need a spinlock

**Review by cd9 — APPROVED** [review]

**Pull request merged**